kafka-rest/src/main/java/io/confluent/kafkarest/controllers/SchemaManagerImpl.java

```diff
@@ -264,4 +321,12 @@ private String getSchemaSubjectUnsafe(
 
     return subject;
   }
+
+  private static void checkArgumentWrapper(boolean argument) {
+    try {
+      checkArgument(argument);
+    } catch (IllegalArgumentException e) {
+      throw new BadRequestException("Unsupported argument: ", e.getMessage(), e);
+    }
```

Comment:
Just reimplement it, i.e.:

```
if (!argument) {
  throw new BadRequestException(...);
}
```

Also, allow for messages, for us to get more descriptive errors, i.e.:

```
checkArgumentWrapper(!format.isPresent(), "`format` shouldn't be present because X.");
```

Reply:
Follow on - this PR could do with merging :) I'll raise an issue.  

This is no worse than it was before wrt to the messages but I do agree they should be better.

Reply:
https://confluentinc.atlassian.net/browse/KREST-4069